src/lib.rs

```diff
@@ -373,6 +373,10 @@ impl<T: Float> Complex<T> {
         // = p^c e^(−d θ) (cos(c θ + d ln(ρ)) + i sin(c θ + d ln(ρ)))
         // = from_polar(p^c e^(−d θ), c θ + d ln(ρ))
         let (r, theta) = self.to_polar();
+
+        if r.is_zero() {
+            return Self::new(r, r);
+        }
```

Comment:
This is wrong if `exp` is also zero or infinity or NaN.

Reply:
Also if `exp.re` is negative, as that would be dividing by zero.

Reply:
`exp` is the second argument (the exponent), not an angle.

Reply:
ah sorry. My bad, I was just focussing on the line you commented on and totally forget about what we're doing here. You are completely right, this cases should be handled here, too. I'm thinking about something like checking `if r.is_zero() && exp.norm().is_finite() && !exp.norm().is_zero()`, which should work. But maybe there is a simpler and cheaper approach.